nautobot/extras/tests/test_jobs.py

```diff
@@ -606,6 +606,19 @@ def test_job_profiling(self):
         self.assertTrue(profiling_result.exists())
         profiling_result.unlink()
 
+    def test_job_singleton(self):
+        module = "singleton"
+        name = "TestSingletonJob"
+
+        job_class, _ = get_job_class_and_model(module, name, "local")
+        self.assertTrue(job_class.is_singleton)
+        cache.set(job_class.singleton_cache_key, 1)
+        failed_job_result = create_job_result_and_run_job(module, name)
+
+        self.assertEqual(
+            failed_job_result.status, JobResultStatusChoices.STATUS_FAILURE, msg="Duplicate singleton job didn't error."
+        )
```

Comment:
Would be good if possible to also add a test for running the job with `_ignore_singleton[_lock]` and confirm that it then runs and clears the flag.